awaitility/src/main/java/com/jayway/awaitility/core/ConditionAwaiter.java

```diff
@@ -164,6 +165,20 @@ public void uncaughtException(Thread thread, Throwable throwable) {
         }
     }
 
+    private ExecutorService initExecutorService() {
+        return Executors.newSingleThreadExecutor(new ThreadFactory() {
+            public Thread newThread(Runnable r) {
+                Thread thread;
+                if (conditionSettings.hasAlias()) {
+                    thread = new Thread(Thread.currentThread().getThreadGroup(), r, "awaitility[" + conditionSettings.getAlias() + ']');
+                } else {
+                    thread = new Thread(Thread.currentThread().getThreadGroup(), r, "awaitility-thread");
```

Comment:
True. I would be nice if the were numbered somehow (like "awaitility-thread-1" etc). But I wouldn't like to keep state (i.e. a counter) around for this if possible.. What are our options?